components/eam/src/physics/cam/cldera_sai_tracers.F90

```diff
@@ -455,35 +422,51 @@ subroutine cldera_sai_tracers_timestep_tend(state, ptend, dt, ncol)
     real(r8) :: pp                ! midpoint pressure (Pa)
     
     ! for injection source localization
-    real(r8) :: V(ncol, pver)                 ! vertical profile (1/m)
-    real(r8) :: sVk   = 0.0_r8                ! vertical profile sum (1/m)
-    real(r8) :: rr    = 1.0e20                ! great circle distance (m)
-    integer  :: inj_owner                     ! MPI rank owning injection column
-    integer  :: inji                          ! injection column index within chunk
-    integer  :: injc                          ! injection local chunk index
-    real(r8) :: inj_dist                      ! distance of matched col from requested location (m)
-    logical  :: inject                        ! true => this rank contains the injection column
-    integer  :: myrank                        ! rank of this MPI process
-    integer  :: ier                           ! return error status
-
-    ! for mass normalization
-    real(r8) :: A_so2 = 0.0                   ! SO2 normalization (kg m / s)
-    real(r8) :: A_ash = 0.0                   ! ash normalization (kg m / s)
+    real(r8) :: V(pver)                  ! vertical profile (1/m)
+    real(r8) :: sVk = 0.0                ! vertical profile sum (1/m)
+    logical  :: inject                   ! true => this rank contains the injection column
+    integer  :: inj_owner                ! MPI rank owning injection column
+    integer  :: inji                     ! injection column index within chunk
+    integer  :: injc                     ! injection local chunk index
+    integer  :: myrank                   ! rank of this MPI process
+    integer  :: ier                      ! return error status
+    real(r8) :: source_on                ! switch for source term
+
+    ! for tracer mass normalization
+    real(r8) :: A_so2 = 0.0              ! SO2 normalization (kg m / s)
+    real(r8) :: A_ash = 0.0              ! ash normalization (kg m / s)
     
-    ! for mass estimate
-    real(r8) :: area(ncol)                    ! column horizontal area (m^2)
-    real(r8) :: grid_mass(ncol,pver)          ! grid cell mass (kg)
-    real(r8) :: col_mass(ncol,pver)           ! cumulative column mass (kg)
+    ! for mass estimates
+    real(r8) :: area(ncol)               ! column horizontal area (m^2)
+    real(r8) :: grid_mass(ncol,pver)     ! grid cell air mass (kg)
+    real(r8) :: col_mass(ncol)           ! total column air mass (kg)
+    real(r8) :: col_mass_so2(ncol)       ! total column SO2 mass (kg)
```

Comment:
Side question: do you think any of these new vars will ever be interesting to track (in order to find anomalies)?

Reply:
@bartgol I've gone over the details of the design and implementation extensively with @hbrown18 and Christiane; I think we can limit this review to resolving conflicts, but also of course any comments on best practices.

As for your side question, I really don't know. The column masses are essentially encoded in the AOD already, but perhaps it could be useful to look at the column masses of species individually (I only provide a total AOD, while there exist column mass variables for so2, sulfate, ash separately).

The area and grid mass I don't think would be useful. Unless you wanted to use the grid mass to track tracer masses specifically rather than mixing ratios.